aws/resource_aws_wafv2_web_acl_association.go

```diff
@@ -0,0 +1,210 @@
+)
+
+func resourceAwsWafv2WebACLAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsWafv2WebACLAssociationCreate,
+		Read:   resourceAwsWafv2WebACLAssociationRead,
+		Delete: resourceAwsWafv2WebACLAssociationDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				webAclArn, resourceArn, err := resourceAwsWafv2ACLAssociationDecodeId(d.Id())
+				if err != nil {
+					return nil, fmt.Errorf("error reading resource ID: %s", err)
+				}
+				d.Set("resource_arn", resourceArn)
+				d.Set("web_acl_arn", webAclArn)
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_arn": {
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(20, 2048),
+			},
+			"web_acl_arn": {
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(20, 2048),
+			},
+		},
+	}
+}
+
+func resourceAwsWafv2WebACLAssociationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafv2conn
+	resourceArn := d.Get("resource_arn").(string)
+	webAclArn := d.Get("web_acl_arn").(string)
+	params := &wafv2.AssociateWebACLInput{
+		ResourceArn: aws.String(resourceArn),
+		WebACLArn:   aws.String(webAclArn),
+	}
+
+	err := resource.Retry(2*time.Minute, func() *resource.RetryError {
+		var err error
+		_, err = conn.AssociateWebACL(params)
+		if err != nil {
+			if isAWSErr(err, wafv2.ErrCodeWAFInternalErrorException, "AWS WAF couldn’t perform the operation because of a system problem") {
+				return resource.RetryableError(err)
+			}
+			if isAWSErr(err, wafv2.ErrCodeWAFUnavailableEntityException, "AWS WAF couldn’t retrieve the resource that you requested") {
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+
+	if isResourceTimeoutError(err) {
+		_, err = conn.AssociateWebACL(params)
+	}
+	if err != nil {
+		return err
+	}
+	d.SetId(fmt.Sprintf("%s,%s", webAclArn, resourceArn))
+
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"associating"},
+		Target:  []string{"associated"},
+		Refresh: wafv2WebACLAssociationRefreshFunc(conn, webAclArn, resourceArn),
+		Timeout: d.Timeout(schema.TimeoutCreate),
+	}
+
+	log.Printf("[DEBUG] Waiting for WAFV2 Web ACL to associate with resource: %s", resourceArn)
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error waiting for WAFV2 Web ACL to associate with resource: %s", err)
+	}
+
+	return resourceAwsWafv2WebACLAssociationRead(d, meta)
+}
+
+func resourceAwsWafv2WebACLAssociationRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafv2conn
+	webAclArn, resourceArn, err := resourceAwsWafv2ACLAssociationDecodeId(d.Id())
```

Comment:
I believe we can read directly from the resource data here since in `Create` -> `Read`  the values are set and at `Import` time, the logic w/in the Importer's `State` func will store the values for us